File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/results/IsFeatureEnabledResult.java
package io.capawesome.capacitorjs.plugins.posthog.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.posthog.interfaces.Result;

public class IsFeatureEnabledResult implements Result {

    private final boolean enabled;

    public IsFeatureEnabledResult(boolean enabled) {
        this.enabled = enabled;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("enabled", this.enabled);
        return result;
    }
}

File: packages/posthog/ios/Plugin/Classes/Results/IsFeatureEnabledResult.swift
import Foundation
import Capacitor

@objc public class IsFeatureEnabledResult: NSObject, Result {
    private let enabled: Bool

    init(enabled: Bool) {
        self.enabled = enabled
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["enabled"] = enabled
        return result as AnyObject
    }
}
